### PR TITLE
Add approval component with proposal type check and choice formatting

### PR DESCRIPTION
**Pull request by samuveth**

### CODE REVIEW

**Comment by samuveth** [comment]
[Create cast vote component for approval voting](https://trello.com/c/95m6aNHk/95-create-cast-vote-component-for-approval-voting)

**Comment by samuveth** [comment]
[Create cast vote component for approval voting](https://trello.com/c/95m6aNHk/95-create-cast-vote-component-for-approval-voting)

File: src/locales/default.json [hunk 1]
```diff
@@ -196,5 +196,14 @@
     "networks": "network(s)",
     "plugins": "plugin(s)",
     "results": "result(s)"
+  },
+  "voting": {
+    "selectVoting": "Select voting system",
+    "single-choice": "Single choice voting",
+    "approval": "Approval voting",
+    "discription": {
```
Comment (bonustrack) [inline src/locales/default.json:204]:
```suggestion
    "description": {
```

File: src/views/Proposal.vue [hunk 2]
```diff
@@ -90,6 +96,12 @@
             <Icon name="external-link" class="ml-1" />
           </a>
         </div>
+        <div class="mb-1">
+          <b>Voting system</b>
+          <a class="float-right">
```
Comment (bonustrack) [inline src/views/Proposal.vue:101]: This should not be a link

**Review by bonustrack — COMMENTED** [review]
There should be a component `src/components/Block/Voting/SingleChoice.vue`

**Comment by bonustrack** [comment]
@samuveth I think you miss this message, it was a bit hidden: https://github.com/snapshot-labs/snapshot/pull/378#pullrequestreview-666192217
You could keep the component "CastVote" and include Voting/SingleChoice or Voting/Approval depending on the proposal type. CastVote component would hold the block and the "Vote" button logic, and SingleChoice/Approval output the field "choice" in their correct format.

**Comment by bonustrack** [comment]
The block to cast a vote doesnt not appear anymore for me, like here: https://bafybeidokyuedj46dqyhljl5bzhjsava6oqgkorasndmyox5b2re6xukkm.on.fleek.co/#/fabien.eth/proposal/QmZSFxVsSyLGKVuKGQTop4pCujTjLF1w25MRp2WgFRPtRp

**Comment by samuveth** [comment]
> The block to cast a vote doesnt not appear anymore for me, like here: https://bafybeidokyuedj46dqyhljl5bzhjsava6oqgkorasndmyox5b2re6xukkm.on.fleek.co/#/fabien.eth/proposal/QmZSFxVsSyLGKVuKGQTop4pCujTjLF1w25MRp2WgFRPtRp

This is weird, I can't reproduce locally

**Comment by bonustrack** [comment]
@samuveth The issue is on CastVotecomponent, the filename is "Castvote.vue" but should be "CastVote.vue"

**Comment by bonustrack** [comment]
We should not show Cast vote block if the proposal is pending or closed. 
image
I think you can remove "canVote" and just check for proposal type directly. And show the castvote block if proposal.state === 'active' (no need for check timestamp anymore)

**Comment by samuveth** [comment]
> We should not show Cast vote block if the proposal is pending or closed.
> I think you can remove "canVote" and just check for proposal type directly. And show the castvote block if proposal.state === 'active' (no need for check timestamp anymore)

Yeah that's much better, forgot that we have state from hub now.

**Comment by samuveth** [comment]
> @samuveth The issue is on CastVotecomponent, the filename is "Castvote.vue" but should be "CastVote.vue"

By git config wasn't setup for case sensitive files somehow.

**Review by bonustrack — APPROVED** [review]
Awesome!

**Pull request merged**